make-release.sh

```diff
@@ -11,147 +11,221 @@
+    COMMIT_MSG="[prerelease] Remove rc from version"
+    git commit -asm "${COMMIT_MSG}"
+  fi
+  $DRY_RUN git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}" || true
+  git checkout "${BRANCH}"
 
+  # change VERSION file + commit change into ${BASEBRANCH} branch
+  # bump the y digit, if it is a major release
+  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=0.1.x, get BASE=0, NEXT=2
+  NEXT_VERSION_Y="${BASE}.${NEXT}.0"
+  bump_version "${NEXT_VERSION_Y}" "${MAIN_BRANCH}"
+  echo "Prerelease is done"
+}
 
-# derive bugfix branch from version
-BRANCH=${VERSION#v}
-BRANCH=${BRANCH%.*}.x
+release() {
+  echo "[INFO] Starting Release procedure"
+  # derive bugfix branch from version
+  BRANCH=${VERSION#v}
+  BRANCH=${BRANCH%.*}.x
 
-# if doing a .0 release, use main branch; if doing a .z release, use $BRANCH
-if [[ ${VERSION} == *".0" ]]; then
-  BASEBRANCH="${MAIN_BRANCH}"
-else
   BASEBRANCH="${BRANCH}"
-fi
 
-# work in tmp dir
-if [[ $TMP ]] && [[ -d $TMP ]]; then
-  pushd "$TMP" > /dev/null || exit 1
+  git remote show origin
+
   # get sources from ${BASEBRANCH} branch
-  echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
-  git clone "${REPO}" -q
-  cd "${REPO##*/}" || exit 1
-fi
+  git fetch origin "${BASEBRANCH}":"${BASEBRANCH}" || true
+  git checkout "${BASEBRANCH}"
+
+  # create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+  if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+    git branch "${BRANCH}" || git checkout "${BRANCH}"
+    $DRY_RUN git push origin "${BRANCH}"
+    git fetch origin "${BRANCH}:${BRANCH}" || true
+    git checkout "${BRANCH}"
+  else
+    git fetch origin "${BRANCH}:${BRANCH}" || true
+    git checkout "${BRANCH}"
+  fi
 
-git remote show origin
+  # change version/version.go file
+  sed -i version/version.go -r -e "s#(Version = \")(v?[0-9.]+)(\")#\1${VERSION}\3#g"
+  # change image tag in Makefile
+  DWO_QUAY_IMG="${DWO_QUAY_REPO}:${VERSION}"
+  sed -i Makefile -r -e "s#quay.io/devfile/devworkspace-controller:[0-9a-zA-Z._-]+#${DWO_QUAY_IMG}#g"
+  docker build -t "${DWO_QUAY_IMG}" -f ./build/Dockerfile .
+  $DRY_RUN docker push "${DWO_QUAY_IMG}"
+
+  PROJECT_CLONE_QUAY_IMG="${PROJECT_CLONE_QUAY_REPO}:${VERSION}"
+  sed -i Makefile -r -e "s#quay.io/devfile/project-clone:.*[0-9a-zA-Z._-]#${PROJECT_CLONE_QUAY_IMG}#g"
+  docker build -t "${PROJECT_CLONE_QUAY_IMG}" -f ./project-clone/Dockerfile .
+  $DRY_RUN docker push "${PROJECT_CLONE_QUAY_IMG}"
+
+  bash -x ./deploy/generate-deployment.sh \
+    --use-defaults \
+    --default-image "${DWO_QUAY_IMG}" \
+    --project-clone-image "${PROJECT_CLONE_QUAY_IMG}"
+
+  # tag the release if the version/version.go file has changed
+  if [[ ! -z $(git status -s) ]]; then # dirty
+    COMMIT_MSG="[release] Release ${VERSION}"
+    git add version/version.go
+    git commit -asm "${COMMIT_MSG}"
```

Comment:
`git add version/version.go` is redundant from the `-a` flag in `git commit`, however using the `-a` flag has slightly different semantics from `git add .` as untracked files are ignored.

```suggestion
    git add .
    git commit -sm "${COMMIT_MSG}"
```

Reply:
now we do `git add -A` and git commit -m "" to do stuff more explicit + not to lose some changes we make but not commit.